akka-bbb-apps/src/main/scala/org/bigbluebutton/core/BigBlueButtonActor.scala

```diff
@@ -117,7 +117,13 @@ class BigBlueButtonActor(
       case None =>
         log.info("Create meeting request. meetingId={}", msg.body.props.meetingProp.intId)
 
-        val m = RunningMeeting(msg.body.props, outGW, eventBus)
+        // Check if we are keeping the events
+        var props =  msg.body.props.copy()
+        if (props.recordProp.keepEvents) {
+          val recordProp = msg.body.props.recordProp.copy(record = true)
+          props = msg.body.props.copy(recordProp = recordProp)

```

Comment:
This sets the meeting as being recorded which would show the record interface on the client. Doesn't it?

I suggest to not override the record boolean but instead pass the keepEvent flag into the OutMsgRouter and let that decide to record the message or not.

https://github.com/bigbluebutton/bigbluebutton/blob/master/akka-bbb-apps/src/main/scala/org/bigbluebutton/core/running/OutMsgRouter.scala#L9

https://github.com/bigbluebutton/bigbluebutton/blob/master/akka-bbb-apps/src/main/scala/org/bigbluebutton/core/running/RunningMeeting.scala#L47

This way the client doesn't have to know if the server is recording the events or not as it is not the client's concern,

Reply:
But I agree this is a little confusing. And I believe my test worked fine in the HTML5 client because Meteor saves everything in mongo... I'll make some changes. Thanks @ritzalam